test/integration/src/test/java/io/pravega/test/integration/KeyValueTableTest.java

```diff
@@ -192,6 +261,1427 @@ public void testCreateDeleteKeyValueTable() {
+        log.info("Successfully completed test Conditional delete single Key using its version");
+    }
+
+    /**
+     * Test case-46: Get Keys version List
+     */
+    @Test
+    public void testGetKeyVersionList() {
+        KeyValueTableInfo newKVT = newKeyValueTableName();
+        KeyValueTable<Integer, String> newKeyValueTable = getKeyValueTable(newKVT);
+        String keyFamily = "KeyFamily1";
+        List<TableEntry<Integer, String>> entryList = new ArrayList<>();
+        Integer[] keyArray = new Integer[10];
+        for (int i = 0; i < keyArray.length; i++) {
+            keyArray[i] = getKeyID();
+        }
+        String[] valueArray = new String[10];
+        for (int i = 0; i < valueArray.length; i++) {
+            valueArray[i] = getValue();
+        }
+
+        // Adding 10 new entries to the Table only if they are not already present.
+        for (int i = 0; i < keyArray.length; i++) {
+            entryList.add(TableEntry.notExists(keyArray[i], valueArray[i]));
+        }
+        log.info("Entry list: {}", entryList);
+        List<Version> versionList = newKeyValueTable.replaceAll(keyFamily, entryList).join();
+        assertNotNull("Version list should not be empty", versionList);
+        assertEquals("Version list size should be same as entry list size",
+                versionList.size(), entryList.size());
+        log.info("Version list: {}", versionList);
+
+        List<Integer> keyList = new ArrayList<>();
+        for (int i = 0; i < keyArray.length; i++) {
+            keyList.add(keyArray[i]);
+        }
+        log.info("Key list: {}", keyList);
+        List<TableEntry<Integer, String>> getEntryList = newKeyValueTable.getAll(keyFamily, keyList).join();
+        assertNotNull("Get Entry List should not be empty", getEntryList);
+        assertEquals("Get Entry List size should be same as get entry list size",
+                getEntryList.size(), entryList.size());
+        log.info("Get Entry List: {}", getEntryList);
+        for (int i = 0; i < keyArray.length; i++) {
+            assertEquals("Corresponding key for getEntryList should be as inserted",
+                    keyList.get(i), getEntryList.get(i).getKey().getKey());
+            assertEquals("Corresponding key for entryList should be as inserted",
+                    entryList.get(i).getKey().getKey(), getEntryList.get(i).getKey().getKey());
+        }
+
+        // Comparing version values of both obtained versionList and getEntryList.
+        for (int i = 0; i < keyArray.length; i++) {
+            assertEquals("All version values of both versionList and getEntryList should be equal",
+                    versionList.get(i), getEntryList.get(i).getKey().getVersion());
+        }
+        log.info("Successfully completed test Get Key versions List");
+    }
+
+    /**
+     * Test case-47: Conditional insert multiple table entries with KeyFamily
+     */
+    @Test
+    public void testConditionInsertMultiKeys() {
```

Comment:
This is already covered in the base class.

Reply:
Sure, I'll remove this test as its a duplicate.